src/core/schema.js

```diff
@@ -138,6 +138,7 @@ function parseProperty (value, propDefinition) {
   // Use default value if value is falsy.
   if (value === undefined || value === null || value === '') {
     value = propDefinition.default;
+    if (Array.isArray(value)) { value = value.slice(); }
```

Comment:
In case I define my own property type and set the default value as an object. I often create "schemaless" components that just accept flexible data.

```
default: {},
parse: function (val) {
  // ... Parse to object or pass through object.
}
```